### DISCUSSION TITLE
USB camera

### DISCUSSION THREAD

**Discussion by jltPITTman** [paukstelis/octoprint_deploy#109] [General]
Connected camera but know picture any suggestions, everything else works great
Thanks

**Comment by paukstelis** [comment]
Use the diagnostic output feature from the utility menu and provide the output.

**Reply by jltPITTman** [reply]
octoprint_deploy diagnostic information. Please provide ALL output for support help
**************************************
/etc/octoprint_deploy
**************************************
octoexec: /home/pi/oprint/bin/octoprint
octopip: /home/pi/oprint/bin/pip
haproxy: true
octopi: true
streamer: camera-streamer
**************************************
/etc/octoprint_instances
**************************************
instance:octoprint port:5000 udev:true
instance:Vyper port:5001 udev:true
instance:Kobra port:5002 udev:true
**************************************
/etc/octoprint_cameras
**************************************
camera:cam_Vyper port:8001 udev:true
camera:cam_Kobra port:8002 udev:true
**************************************
/etc/udev/rules.d/99-octoprint.rules
**************************************
KERNELS=="1-1.1:1.0",SUBSYSTEM=="tty",SYMLINK+="octo_octoprint"
KERNELS=="1-1.1:1.0",SUBSYSTEM=="tty",SYMLINK+="octo_Vyper"
SUBSYSTEM=="video4linux", ATTRS{serial}=="84E1F2AF", ATTR{index}=="0", SYMLINK+="cam_Vyper"
KERNELS=="1-1.2:1.0",SUBSYSTEM=="tty",SYMLINK+="octo_Kobra"
SUBSYSTEM=="video4linux", ATTRS{serial}=="SN0001", ATTR{index}=="0", SYMLINK+="cam_Kobra"
lrwxrwxrwx 1 root root 7 Jul 31 00:17 /dev/octo_Kobra -> ttyUSB1
lrwxrwxrwx 1 root root 7 Jul 31 00:17 /dev/octo_octoprint -> ttyUSB0
lrwxrwxrwx 1 root root 7 Jul 31 00:17 /dev/octo_Vyper -> ttyUSB0
lrwxrwxrwx 1 root root 6 Jul 31 00:17 /dev/cam_Kobra -> video2
lrwxrwxrwx 1 root root 6 Jul 31 00:17 /dev/cam_Vyper -> video0
**************************************
Unit octoprint.service could not be found.
None
None
{'bitrate': '10000k',
 'cleanTmpAfterDays': 7,
 'defaultWebcam': 'classic',
 'ffmpeg': None,
 'ffmpegCommandline': '{ffmpeg} -framerate {fps} -i "{input}" -vcodec '
                      '{videocodec} -threads {threads} -b:v {bitrate} -f '
                      '{containerformat} -y {filters} "{output}"',
 'ffmpegThreads': 1,
 'ffmpegThumbnailCommandline': '{ffmpeg} -sseof -1 -i "{input}" -update 1 -q:v '
                               '0.7 "{output}"',
 'ffmpegVideoCodec': 'libx264',
 'snapshotWebcam': 'classic',
 'timelapse': {'fps': 25,
               'options': {'capturePostRoll': None,
                           'interval': None,
                           'retractionZHop': None},
               'postRoll': 0,
               'type': 'off'},
 'timelapseEnabled': True,
 'watermark': True,
 'webcamEnabled': True}
**************************************
● Vyper.service - The snappy web interface for your 3D printer
     Loaded: loaded (/etc/systemd/system/Vyper.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2023-07-31 21:54:44 BST; 5min ago
   Main PID: 1304 (octoprint)
      Tasks: 37 (limit: 4915)
        CPU: 54.984s
     CGroup: /system.slice/Vyper.service
             └─1304 /home/pi/oprint/bin/python /home/pi/oprint/bin/octoprint serve --config=/home/pi/.Vyper/config.yaml --basedir=/home/pi/.Vyper --port=5001

Jul 31 21:55:52 octopi octoprint[1304]: 2023-07-31 21:55:52,976 - octoprint.plugins.firmware_check - INFO - Your printer's firmware is a development build of Marlin (build date 20210902). It might be more unstable than a release version and should be kept up-to-date.. More information at https://faq.octoprint.org/warning-firmware-development
Jul 31 21:55:53 octopi octoprint[1304]: 2023-07-31 21:55:53,007 - octoprint.util.comm - INFO - Firmware states that it supports temperature autoreporting
Jul 31 21:55:53 octopi octoprint[1304]: 2023-07-31 21:55:53,061 - octoprint.util.comm - INFO - Firmware states that it supports emergency GCODEs to be sent without waiting for an acknowledgement first
Jul 31 21:55:53 octopi octoprint[1304]: 2023-07-31 21:55:53,214 - octoprint.server - INFO - Autorefresh of serial port list stopped
Jul 31 21:55:53 octopi octoprint[1304]: 2023-07-31 21:55:53,291 - octoprint.util.comm - WARNING - Got a file from printer's SD that has a non-ascii filename ('/×ÊÁÏ_~1/OWL~1.GCO'), that shouldn't happen according to the protocol
Jul 31 21:55:53 octopi octoprint[1304]: 2023-07-31 21:55:53,416 - octoprint.util.comm - WARNING - Got a file from printer's SD that has a non-ascii filename ('/×ÊÁÏ_~1/OWL~1.GCO'), that shouldn't happen according to the protocol
Jul 31 21:55:53 octopi octoprint[1304]: 2023-07-31 21:55:53,439 - octoprint.plugins.tracking - INFO - Sent tracking event printer_connected, payload: {'firmware_name': 'Marlin bugfix-2.0.x (Sep  2 2021 20:16:33)', 'printer_port': '/dev/octo_Vyper', 'printer_baudrate': 0}
Jul 31 21:56:02 octopi octoprint[1304]: 2023-07-31 21:56:02,418 - octoprint.server.util.flask - INFO - Passively logging in user jltpittman from 127.0.0.1
Jul 31 21:56:02 octopi octoprint[1304]: 2023-07-31 21:56:02,419 - octoprint.access.users - INFO - Logged in user: jltpittman
Jul 31 21:56:03 octopi octoprint[1304]: 2023-07-31 21:56:03,085 - octoprint.plugins.octoeverywhere - INFO - Slipstream took 0.6844155788421631 to fully update the cache
None
None
{'bitrate': '10000k',
 'cleanTmpAfterDays': 7,
 'defaultWebcam': 'classic',
 'ffmpeg': None,
 'ffmpegCommandline': '{ffmpeg} -framerate {fps} -i "{input}" -vcodec '
                      '{videocodec} -threads {threads} -b:v {bitrate} -f '
                      '{containerformat} -y {filters} "{output}"',
 'ffmpegThreads': 1,
 'ffmpegThumbnailCommandline': '{ffmpeg} -sseof -1 -i "{input}" -update 1 -q:v '
                               '0.7 "{output}"',
 'ffmpegVideoCodec': 'libx264',
 'snapshotWebcam': 'classic',
 'timelapse': {'fps': 25,
               'options': {'capturePostRoll': None,
                           'interval': None,
                           'retractionZHop': None},
               'postRoll': 0,
               'type': 'off'},
 'timelapseEnabled': True,
 'watermark': True,
 'webcamEnabled': True}
**************************************
● Kobra.service - The snappy web interface for your 3D printer
     Loaded: loaded (/etc/systemd/system/Kobra.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2023-07-31 21:54:41 BST; 5min ago
   Main PID: 1249 (octoprint)
      Tasks: 37 (limit: 4915)
        CPU: 34.343s
     CGroup: /system.slice/Kobra.service
             └─1249 /home/pi/oprint/bin/python /home/pi/oprint/bin/octoprint serve --config=/home/pi/.Kobra/config.yaml --basedir=/home/pi/.Kobra --port=5002

Jul 31 21:55:50 octopi octoprint[1249]: 2023-07-31 21:55:50,418 - octoprint.server - INFO - Autorefresh of serial port list stopped
Jul 31 21:55:50 octopi octoprint[1249]: 2023-07-31 21:55:50,508 - octoprint.util.comm - INFO - Printer reports firmware name "Marlin bugfix-2.0.x (Dec 22 2021 14:30:26)"
Jul 31 21:55:50 octopi octoprint[1249]: 2023-07-31 21:55:50,512 - octoprint.plugins.firmware_check - INFO - Your printer's firmware is a development build of Marlin (build date 20211222). It might be more unstable than a release version and should be kept up-to-date.. More information at https://faq.octoprint.org/warning-firmware-development
Jul 31 21:55:50 octopi octoprint[1249]: 2023-07-31 21:55:50,579 - octoprint.util.comm - INFO - Firmware states that it supports temperature autoreporting
Jul 31 21:55:50 octopi octoprint[1249]: 2023-07-31 21:55:50,615 - octoprint.util.comm - INFO - Firmware states that it supports emergency GCODEs to be sent without waiting for an acknowledgement first
Jul 31 21:55:51 octopi octoprint[1249]: 2023-07-31 21:55:51,035 - octoprint.plugins.tracking - INFO - Sent tracking event printer_connected, payload: {'firmware_name': 'Marlin bugfix-2.0.x (Dec 22 2021 14:30:26)', 'printer_port': '/dev/octo_Kobra', 'printer_baudrate': 0}
Jul 31 21:59:45 octopi octoprint[1249]: 2023-07-31 21:59:45,562 - octoprint.plugins.octoeverywhere - INFO - Starting a secondary connection to wss://tor.octoeverywhere.com/octoclientws method 2
Jul 31 21:59:45 octopi octoprint[1249]: 2023-07-31 21:59:45,566 - octoprint.plugins.octoeverywhere - INFO - Attempting to talk to OctoEverywhere, server con wss://tor.octoeverywhere.com/octoclientws[1] wsId:2701683656
Jul 31 21:59:45 octopi octoprint[1249]: 2023-07-31 21:59:45,863 - octoprint.plugins.octoeverywhere - INFO - Connected To OctoEverywhere, server con wss://tor.octoeverywhere.com/octoclientws[1]. Starting handshake...
Jul 31 21:59:45 octopi octoprint[1249]: 2023-07-31 21:59:45,969 - octoprint.plugins.octoeverywhere - INFO - Handshake complete, server con wss://tor.octoeverywhere.com/octoclientws[1], successfully connected to OctoEverywhere!
None
None
{'bitrate': '10000k',
 'cleanTmpAfterDays': 7,
 'defaultWebcam': 'classic',
 'ffmpeg': None,
 'ffmpegCommandline': '{ffmpeg} -framerate {fps} -i "{input}" -vcodec '
                      '{videocodec} -threads {threads} -b:v {bitrate} -f '
                      '{containerformat} -y {filters} "{output}"',
 'ffmpegThreads': 1,
 'ffmpegThumbnailCommandline': '{ffmpeg} -sseof -1 -i "{input}" -update 1 -q:v '
                               '0.7 "{output}"',
 'ffmpegVideoCodec': 'libx264',
 'snapshotWebcam': 'classic',
 'timelapse': {'fps': 25,
               'options': {'capturePostRoll': None,
                           'interval': None,
                           'retractionZHop': None},
               'postRoll': 0,
               'type': 'off'},
 'timelapseEnabled': True,
 'watermark': True,
 'webcamEnabled': True}
**************************************
Unit cam_Vyper.service could not be found.
**************************************
Unit cam_Kobra.service could not be found.
**************************************
● haproxy.service - HAProxy Load Balancer
     Loaded: loaded (/lib/systemd/system/haproxy.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2023-07-31 21:51:07 BST; 9min ago
       Docs: man:haproxy(1)
             file:/usr/share/doc/haproxy/configuration.txt.gz
    Process: 974 ExecStartPre=/usr/sbin/haproxy -f $CONFIG -c -q $EXTRAOPTS (code=exited, status=0/SUCCESS)
   Main PID: 985 (haproxy)
      Tasks: 5 (limit: 4915)
        CPU: 582ms
     CGroup: /system.slice/haproxy.service
             ├─985 /usr/sbin/haproxy -Ws -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid -S /run/haproxy-master.sock
             └─989 /usr/sbin/haproxy -Ws -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid -S /run/haproxy-master.sock

Jul 31 21:55:50 octopi haproxy[989]: ::ffff:192.168.1.21:58394 [31/Jul/2023:21:55:50.045] public Kobra/octoprint1 2/0/0/209/211 204 887 - - ---- 8/8/1/1/0 0/0 "POST /Kobra/api/connection HTTP/1.1"
Jul 31 21:55:51 octopi haproxy[989]: ::ffff:192.168.1.21:58396 [31/Jul/2023:21:55:48.047] public octoprint/octoprint1 0/3006/-1/-1/3006 503 3668 - - SC-- 8/8/1/0/3 0/0 "GET /static/img/tentacle-20x20.png HTTP/1.1"
Jul 31 21:55:51 octopi haproxy[989]: ::ffff:192.168.1.21:58400 [31/Jul/2023:21:55:48.062] public octoprint/octoprint1 0/3005/-1/-1/3005 503 3668 - - SC-- 7/7/0/0/3 0/0 "GET /plugin/dashboard/static/img/command-icon.png HTTP/1.1"
Jul 31 21:55:51 octopi haproxy[989]: ::ffff:192.168.1.21:58394 [31/Jul/2023:21:55:50.271] public Kobra/octoprint1 3/0/0/1579/1582 200 34479 - - ---- 6/6/2/2/0 0/0 "GET /Kobra/api/settings HTTP/1.1"
Jul 31 21:55:51 octopi haproxy[989]: ::ffff:192.168.1.21:58422 [31/Jul/2023:21:55:50.274] public Kobra/octoprint1 0/0/0/1630/1630 200 1982 - - ---- 6/6/1/1/0 0/0 "GET /Kobra/api/printerprofiles HTTP/1.1"
Jul 31 21:55:52 octopi haproxy[989]: ::ffff:192.168.1.21:58394 [31/Jul/2023:21:55:51.964] public Kobra/octoprint1 3/0/1/44/48 200 1380 - - ---- 6/6/4/4/0 0/0 "GET /Kobra/api/connection HTTP/1.1"
Jul 31 21:55:52 octopi haproxy[989]: ::ffff:192.168.1.21:58425 [31/Jul/2023:21:55:51.969] public Kobra/octoprint1 9/0/0/51/60 200 1104 - - ---- 6/6/3/3/0 0/0 "GET /Kobra/api/plugin/firmware_check HTTP/1.1"
Jul 31 21:55:52 octopi haproxy[989]: ::ffff:192.168.1.21:58402 [31/Jul/2023:21:55:51.982] public Kobra/octoprint1 0/0/0/58/58 304 883 - - ---- 6/6/2/2/0 0/0 "GET /Kobra/plugin/DisplayLayerProgress/static/font/DOTMATRI.woff HTTP/1.1"
Jul 31 21:55:52 octopi haproxy[989]: ::ffff:192.168.1.21:58401 [31/Jul/2023:21:55:52.006] public Kobra/octoprint1 0/0/1/53/54 200 18567 - - ---- 6/6/1/1/0 0/0 "GET /Kobra/api/files?recursive=true HTTP/1.1"
Jul 31 21:55:54 octopi haproxy[989]: ::ffff:192.168.1.21:58422 [31/Jul/2023:21:55:51.945] public octoprint/octoprint1 0/3004/-1/-1/3004 503 3668 - - SC-- 2/2/0/0/3 0/0 "GET /plugin/dashboard/static/img/command-icon.png HTTP/1.1"

**Comment by paukstelis** [comment]
I'm not sure how you have camera-streamer setup as your option, since I removed that some time ago.

**Comment by jltPITTman** [comment]
Suggest anything to try different

**Comment by paukstelis** [comment]
Using the menu, remove all the cameras. Using the utility menu, change the streamer to ustreamer. Then create the cameras again.

**Reply by jltPITTman** [reply]
Thanks I'll try it great job on creating this
